Utilities/Remotely_Server_Install.sh

```diff
@@ -2,49 +2,84 @@
 fi
 
+echo
+echo "Enter a server host. If you're using a LAN, put the IP Address from the server host. For WAN, put the domain name."
 read -p "Enter server host (e.g. remotely.yourdomainname.com): " serverHost
 
-UbuntuVersion=$(lsb_release -r -s)
-
-apt-get update
+if [ -f /etc/os-release ]; then
+    . /etc/os-release
+    OS=$NAME
+    VER=$VERSION_ID
+elif type lsb_release >/dev/null 2>&1; then
+    OS=$(lsb_release -si)
+    VER=$(lsb_release -sr)
+elif [ -f /etc/lsb-release ]; then
+    . /etc/lsb-release
+    OS=$DISTRIB_ID
+    VER=$DISTRIB_RELEASE
+elif [ -f /etc/debian_version ]; then
+    OS=Debian
+    VER=$(cat /etc/debian_version)
+fi
 
+echo "-- Installing .NET Core and Dependencies..."
+{
 # Install .NET Core Runtime.
-wget -q https://packages.microsoft.com/config/ubuntu/$UbuntuVersion/packages-microsoft-prod.deb
+if [ -f "./packages-microsoft-prod.deb" ]; then
+        echo "File downloaded. Not needed to download"
+elif [ "$OS" = "Ubuntu" ]; then
+        echo "Ubuntu OS Detected"
+        wget -q https://packages.microsoft.com/config/ubuntu/$VER/packages-microsoft-prod.deb
+elif [ "$OS" = "Debian GNU/Linux" ] && [ "$VER" = "9" ]; then
+        echo "Debian OS 9 Detected"
+        wget -O - https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > microsoft.asc.gpg
+        sudo mv microsoft.asc.gpg /etc/apt/trusted.gpg.d/
+        wget https://packages.microsoft.com/config/debian/$VER/prod.list
+        sudo mv prod.list /etc/apt/sources.list.d/microsoft-prod.list
+        sudo chown root:root /etc/apt/trusted.gpg.d/microsoft.asc.gpg
+        sudo chown root:root /etc/apt/sources.list.d/microsoft-prod.list
+elif [ "$OS" = "Debian GNU/Linux" ] && [ "$VER" = "10" ]; then
+	echo "Debian OS 10 Detected"
+        wget https://packages.microsoft.com/config/debian/10/packages-microsoft-prod.deb
+fi
 dpkg -i packages-microsoft-prod.deb
+rm packages-microsoft-prod.deb
 add-apt-repository universe
-apt-get -y install apt-transport-https
 apt-get update
-apt-get -y install aspnetcore-runtime-3.1
-rm packages-microsoft-prod.deb
-
-
- # Install other prerequisites.
-apt-get -y install unzip
-apt-get -y install acl
-apt-get -y install libc6-dev
-apt-get -y install libgdiplus
+apt-get -y install apt-transport-https aspnetcore-runtime-3.1
 
+ # Install dependencies
+apt-get -y install unzip acl libc6-dev libgdiplus nginx
+} &> /dev/null
 
 # Download and install Remotely files.
+echo "-- Installing Remotely Server..."
+{
+if [ ! -f "./Remotely_Server_Linux-x64.zip" ]; then
+	wget "https://github.com/Jay-Rad/Remotely/releases/latest/download/Remotely_Server_Linux-x64.zip"
+fi
 mkdir -p $appRoot
-wget "https://github.com/Jay-Rad/Remotely/releases/latest/download/Remotely_Server_Linux-x64.zip"
 unzip -o Remotely_Server_Linux-x64.zip -d $appRoot
-rm Remotely_Server_Linux-x64.zip
+rm ./Remotely_Server_Linux-x64.zip
 setfacl -R -m u:www-data:rwx $appRoot
 chown -R www-data:www-data $appRoot
+} &> /dev/null
```

Comment:
Not necesary but recommended. Use a ./ for files that in the working directory. It can be a headache when remove or install files from other folders.